docs/logging-requests.md

```diff
@@ -0,0 +1,175 @@
+
+        public LoggingHandler(ILogger<LoggingHandler> logger) {
+            _logger = logger;
+        }
+
+        /// <summary>
+        /// Sends a HTTP request.
+        /// </summary>
+        /// <param name="httpRequest">The <see cref="HttpRequestMessage"/> to be sent.</param>
+        /// <param name="cancellationToken">The <see cref="CancellationToken"/> for the request.</param>
+        /// <returns></returns>
+        protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage httpRequest, CancellationToken cancellationToken)
+        {
+            _logger.LogInformation("Sending Graph request", httpRequest);// log the request before it goes out.
+            HttpResponseMessage response = await base.SendAsync(httpRequest, cancellationToken);
+            _logger.LogInformation("Received Graph response", response);// log the response as it comes back.
+            return response;
+        }
+    }
+}
+```
+
+### Step 2: Add the Handler to GraphServiceClient's list of handlers
+
+Once the logging handler is implemented. You can add it to the list of handlers used by the SDK as follows.
+
+```cs
+// create the auth provider
+var authenticationProvider = new TokenCredentialAuthProvider(clientCertificateCredential,scopes);
+
+// get the default list of handlers and add the logging handler to the list
+var handlers = GraphClientFactory.CreateDefaultHandlers(authenticationProvider);
+handlers.Add(new LoggingHandler());
+
+// create the GraphServiceClient with logging support
+var httpClient = GraphClientFactory.Create(handlers);
+GraphServiceClient graphServiceClient = new GraphServiceClient(httpClient);
+
+// make a request with logging enabled!!
+User me = await graphServiceClient.Me.Request().GetAsync();
+```
+
+## b. Take advantage of OpenTelemetry's instrumentation of HttpClient
+
+Since the GraphServiceClient uses the native HttpClient internally, it is possible to take advantage of the DiagnositicHandler that is present in the default [HttpClientHandler](https://github.com/dotnet/runtime/blob/766fec7d6a9c4fad3d1f44bfe9ec2733c6689ac8/src/libraries/System.Net.Http/src/System/Net/Http/HttpClientHandler.cs#L33).
+
+This can be done using the following steps.
+
+### Step 1: Install OpenTelemetry.Instrumentation.Http Package
+
+Add the reference to [`OpenTelemetry.Instrumentation.Http`](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.Http) to your project. Also, add any other instrumentations & exporters you will need.
+
+```shell
+dotnet add package OpenTelemetry.Instrumentation.Http
+```
+
+### Step 2: Enable HTTP Instrumentation in your application.
+
+This example also sets up the OpenTelemetry Console exporter, which requires adding the package
+[`OpenTelemetry.Exporter.Console`](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.Console/README.md)
+to the application.
+
+```cs
+using var tracerProvider = Sdk.CreateTracerProviderBuilder()
```

Comment:
where does this Sdk symbol come from?

Reply:
This comes from the `OpenTelemetry.Instrumentation.Http` package to be installed in Step 1

Reply:
ok, thanks for the information. It's "strange" they chose such a generic name for the entry point.
As the snippet doesn't feature usings, maybe add a comment that says Sdk comes from "namespace name"? 
This is a question that comes often in samples "where does this thing come from?"

Reply:
That is valid. Will action and resolve.